### DISCUSSION TITLE
Fixing the issue with module resolution in typescript

### DISCUSSION THREAD

**Discussion by rishavs** [KilledByAPixel/LittleJS#87] [General]
If you are using any typescript tooling like vite, Bun, esbuild etc, very likely you would have hit the issue of module resolution not working for the littlejsengine types.

To get around this you can do the following;
1. locate the `littlejs.d.ts` file in the `node_modules\littlejsengine\build\` folder
2. copy it to `node_modules\littlejsengine` folder and rename to `index.d.ts`
3. in the littlejs.d.ts file, change the first line to `declare module "littlejsengine"` so that the module name is the same as the npm package name
4. now, your code should be able to read the types correctly.

**Comment by KilledByAPixel** [comment]
This should be fixed now! There is code to fix up the module name so it should work. Let me know if you have any trouble.

https://github.com/KilledByAPixel/LittleJS/blob/10073e056171beb3fe45b64bbfcbe94b4a563759/dist/littlejs.d.ts#L1
